File: assets/samples/domain.availability/apikey/net/DomainAvailabilityApi.cs
using System;
using System.IO;
using System.Net;
using System.Security.Cryptography;
using System.Text;

class Program {
    public const string domain = "example.com";
    public const string key = "your domain availability api key";
    public const string secret = "your domain availability api secret key";
    public const string username = "your domain availability api username";

    static void Main() {
        long time = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))
                            .TotalMilliseconds);
        string reqData='\u007B'+$"\"t\":{time},\"u\":\"{username}\""+'\u007D';
        var reqBytes = Encoding.UTF8.GetBytes(reqData);
        string req = Convert.ToBase64String(reqBytes);

        string data = username + time + key;
        HMACMD5 hmac = new HMACMD5(Encoding.UTF8.GetBytes(secret));
        string hex = BitConverter.ToString(
            hmac.ComputeHash(Encoding.UTF8.GetBytes(data)));
        string digest = hex.Replace("-", "").ToLower();

        string url="http://www.whoisxmlapi.com/whoisserver/WhoisService?"
                  +$"requestObject={req}&digest={digest}&domainName={domain}"
                  +"&cmd=GET_DN_AVAILABILITY";
        string whoisData = string.Empty;

        HttpWebRequest rq = (HttpWebRequest)WebRequest.Create(url);

        using (HttpWebResponse response = (HttpWebResponse)rq.GetResponse())
        using (Stream stream = response.GetResponseStream())
        using (StreamReader reader = new StreamReader(stream)) {
            whoisData = reader.ReadToEnd();
        }
        Console.WriteLine(whoisData);
    }
}

File: assets/samples/domain.availability/apikey/java/DomainAvailabilityApi.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import org.apache.commons.codec.binary.Base64;
import org.apache.commons.codec.binary.Hex;

public class DomainAvailabilityApi {
    public static final String DOMAIN = "example.com";
    public static final String KEY = "your domain availability api key";
    public static final String SECRET =
        "your domain availability api secret key";
    public static final String USERNAME =
        "your domain availability api username";

    public static void main(String[] args) throws Exception {
        long time = System.currentTimeMillis();

        String json = "{\"t\":" + time + ",\"u\":\"" + USERNAME + "\"}";
        String req=new String(new Base64(true).encodeBase64(json.getBytes()));

        String algo = "HmacMD5";
        String data = USERNAME + time + KEY;
        SecretKeySpec spec = new SecretKeySpec(SECRET.getBytes("UTF-8"),algo);
        Mac mac = Mac.getInstance(spec.getAlgorithm());
        mac.init(spec);
        String hmac =
            new String(Hex.encodeHex(mac.doFinal(data.getBytes("UTF-8"))));

        String url = "http://www.whoisxmlapi.com/whoisserver/WhoisService"
                   + "?requestObject=" + req + "&digest=" + hmac
                   + "&domainName=" + DOMAIN + "&cmd=GET_DN_AVAILABILITY"
                   + "&getMode=DNS_AND_WHOIS";

        try (java.util.Scanner s =
                new java.util.Scanner(new java.net.URL(url).openStream())) {
            System.out.println(s.useDelimiter("\\A").next());
        }
    }
}